src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/CommandLineItemHandler.cs

```diff
@@ -53,9 +53,11 @@ public override Task HandleAsync(IProjectVersionedValue<IProjectSubscriptionUpda
             if (!ProcessDesignTimeBuildFailure(projectChange, context))
             {
                 ProcessOptions(projectChange, context);
-                ProcessItems(projectChange, context, isActiveContext);
             }
 
+            // We need to process items even from failed design time builds, as we might be removing some items.
```

Comment:
Hmm interesting. So, for the cases where the design time build has really failed badly and we never get a second update with added items, we will end up with no intellisense or source files for the project.

In that case probably we need to ask CPS to investigate why they are raising 2 updates for the repro mentioned in #1445? This also seems like a recent CPS regression, because our handling of failed design time builds hasn't changed but we have started running into this issue recently.